sql/field.cc

```diff
@@ -1879,8 +1882,13 @@ Field *Field::new_field(MEM_ROOT *root, TABLE *new_table,
     never be quite sure which parts of the server will break.
   */
   tmp->unireg_check= Field::NONE;
+  /* COMPRESSED column format flag must not be cleared here */
+  bool has_compressed_flag = 
+    (tmp->column_format() == COLUMN_FORMAT_TYPE_COMPRESSED);
   tmp->flags&= (NOT_NULL_FLAG | BLOB_FLAG | UNSIGNED_FLAG |
                 ZEROFILL_FLAG | BINARY_FLAG | ENUM_FLAG | SET_FLAG);
```

Comment:
Originally, other format types (`COLUMN_FORMAT_TYPE_FIXED` and `COLUMN_FORMAT_TYPE_DYNAMIC`) were cleared here. Currently, we need to preserve this flag only for 'COLUMN_FORMAT_TYPE_COMPRESSED'.
As far as I remember this is necessary for 'CREATE TABLE ... AS SELECT * FROM ...' to preserve compression dictionary references in the new table which is still an open question whether we want this behavior or not.